runtime/metricsview/executor_validate.go

```diff
@@ -272,6 +275,47 @@ func (e *Executor) validateIndividualDimensionsAndMeasures(ctx context.Context,
 	slices.SortFunc(res.MeasureErrs, func(a, b IndexErr) int { return a.Idx - b.Idx })
 }
 
+// validateTimeTypes validates the time dimension and dimensions of type timestamp or date in the metrics view.
+func (e *Executor) validateTimeTypes(tableSchema map[string]*runtimev1.StructType_Field, mvSchema map[string]*runtimev1.Type, res *ValidateMetricsViewResult) {
+	// Check if the time dimension is set, if primary time dim not set then other time types cannot be used while querying the metrics view, but that check will be done in the query executor
+	if e.metricsView.TimeDimension == "" {
+		return
+	}
+
+	// Time dimension should either exist in table schema if referring to a column or defined in the metics view
+	var timeTypeCode runtimev1.Type_Code
+	f, ok := tableSchema[strings.ToLower(e.metricsView.TimeDimension)]
+	if !ok {
+		// check if the time dimension is defined in the metrics view schema
+		timeType, ok := mvSchema[e.metricsView.TimeDimension]
+		if !ok {
+			res.TimeDimensionErr = fmt.Errorf("timeseries %q is not a column in table %q or defined in metrics view", e.metricsView.TimeDimension, e.metricsView.Table)
+			return
+		}
+		timeTypeCode = timeType.Code
+	} else {
+		timeTypeCode = f.Type.Code
+	}
+
+	if timeTypeCode != runtimev1.Type_CODE_TIMESTAMP && timeTypeCode != runtimev1.Type_CODE_DATE && !(e.olap.Dialect() == drivers.DialectPinot && timeTypeCode == runtimev1.Type_CODE_INT64) {
+		res.TimeDimensionErr = fmt.Errorf("timeseries %q is not a TIMESTAMP column", e.metricsView.TimeDimension)
+		return
+	}
+
+	// check all dimension of type timestamp or date - if they are referring to a column then needs to use same name as column as the column name can be used in the where time filter without selecting the dimension
+	for i, d := range e.metricsView.Dimensions {
+		if d.Type == nil || (d.Type.Code != runtimev1.Type_CODE_TIMESTAMP && d.Type.Code != runtimev1.Type_CODE_DATE) {
+			continue
+		}
+		if d.Column != "" && d.Column != d.Name {
+			res.DimensionErrs = append(res.DimensionErrs, IndexErr{
+				Idx: i,
+				Err: fmt.Errorf("dimension %q must have the same name as the column as it is of type %s", d.Name, d.Type.Code),
+			})
+		}
```

Comment:
Why can't we allow a different name than the underlying column? At least when the time dimension is explicitly defined

Reply:
I wanted to disallow redefining same time column multiple times but I guess its also allowed for other dimensions so I can remove the restriction.